estate/models/estate_property_offer.py

```diff
@@ -57,3 +58,12 @@ def action_refuse_offer(self):
         self.property_id.write({"selling_price": 0})
         self.write({"status": "refused"})
         return True
+
+    @api.model
+    def create(self, vals):
+        property = self.env["estate.property"].browse(vals["property_id"])
+        max_price = max(property.offer_ids.mapped("price"))
+        if max_price and float_compare(vals["price"], max_price, precision_digits=2) == -1:
+            raise UserError("The offer price cannot be lower than %.2f" % max_price)
+        vals.update({"status": "offer_received"})
+        return super().create(vals)
```

Comment:
Sorry, I was thinking this was the `estate_property` model, so updating the `vals` variable is not okay here, because it's not the property status but the property offer.

```suggestion
        actual_property = self.env["estate.property"].browse(vals["property_id"])
        max_price = max(actual_property.offer_ids.mapped("price"))
        if max_price and float_compare(vals["price"], max_price, precision_digits=2) == -1:
            raise UserError("The offer price cannot be lower than %.2f" % max_price)
        actual_property.status = "offer_received"
        return super().create(vals)
```